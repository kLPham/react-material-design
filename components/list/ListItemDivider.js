import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ListItemDivider = ({ inset }) => <li role="separator" className={classNames('mdc-list-divider', { 'mdc-list-divider--inset': inset })} />;
ListItemDivider.propTypes = {
    inset: PropTypes.bool,
};
export default ListItemDivider;
