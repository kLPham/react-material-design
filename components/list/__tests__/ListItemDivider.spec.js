import React from 'react';
import { shallow } from 'enzyme';
import ListItemDivider from '../ListItemDivider';

describe('<ListItemDivider />', () => {
    it('should render as list item div', () => {
        const wrapper = shallow(
            <ListItemDivider />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-divider');
        expect(wrapper).toHaveProp('role');
    });
    it('should render as list item div with inset', () => {
        const wrapper = shallow(
            <ListItemDivider inset />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveClassName('mdc-list-divider mdc-list-divider--inset');
        expect(wrapper).toHaveProp('role');
    });
});
