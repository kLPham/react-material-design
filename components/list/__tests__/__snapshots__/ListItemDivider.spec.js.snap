exports[`<ListItemDivider /> should render as list item div 1`] = `
<li
  className="mdc-list-divider"
  role="separator" />
`;

exports[`<ListItemDivider /> should render as list item div with inset 1`] = `
<li
  className="mdc-list-divider mdc-list-divider--inset"
  role="separator" />
`;
